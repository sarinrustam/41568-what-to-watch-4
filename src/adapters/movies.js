export const movieAdapter = (movieData) => {
  return {
    id: movieData[`id`],
    title: movieData[`name`],
    poster: movieData[`poster_image`],
    release: movieData[`released`],
    genre: movieData[`genre`],
    isFavorite: movieData[`is_favorite`],
    coverBackground: movieData[`background_image`],
    rating: {
      score: movieData[`rating`],
      scoreDesc: `FUNCTION`,
      amount: movieData[`scores_count`]
    },
    description: movieData[`description`],
    crew: {
      director: movieData[`director`],
      actors: movieData[`starring`]
    },
    preview: movieData[`preview_video_link`],
    videoLink: movieData[`video_link`],
    duration: movieData[`run_time`],
  };
};

export const moviesAdapter = (dataFromBackend) => {
  return dataFromBackend.map(movieAdapter);
};
