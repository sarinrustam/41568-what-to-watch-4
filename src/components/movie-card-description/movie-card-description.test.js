import React from "react";
import renderer from "react-test-renderer";

import MovieCardDescription from './movie-card-description.jsx';

import movies from "../../mocks/films.js";

describe(`MovieCardDescription render`, () => {
  it(`MovieCardDescription render`, () => {
    const handlerSetActive = jest.fn();
    const tree = renderer.create(
        <MovieCardDescription
          movie={movies[0]}
          activeItem={`Overview`}
          setActiveItem={handlerSetActive}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
