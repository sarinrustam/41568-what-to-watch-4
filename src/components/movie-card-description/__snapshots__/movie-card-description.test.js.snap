// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardDescription render MovieCardDescription render 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      2.5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        2323
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Quentin Tarantino
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tim Roth, Amanda Plummer, Laura Lovelace, John Travolta, Samuel L. Jackson
         and other
      </strong>
    </p>
  </div>
</div>
`;
